### ISSUE TITLE
Do not index documents that have not been updated

### ISSUE THREAD

**Issue by dadoonet** [dadoonet/fscrawler#265]
Apparently we reindex systematically all docs for every run.

I looked at `test_add_new_file()` test and it appears that we always index old files:

```
18:48:20,516 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] Running test suite with Locale [ar_QA]
18:48:20,521 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] Running test suite with TimeZone [America/Resolute]/[Central Standard Time]
18:48:20,901 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   --> Test metadata dir ready in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/.fscrawler]
18:48:21,187 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] The cluster is secured. So we need to build a client with security
org.elasticsearch.client.ResponseException: GET http://127.0.0.1:9200/: HTTP/1.1 401 Unauthorized
{"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication token for REST request [/]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}}],"type":"security_exception","reason":"missing authentication token for REST request [/]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}},"status":401}
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:311) ~[rest-5.1.1.jar:5.1.1]
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:300) ~[rest-5.1.1.jar:5.1.1]
	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119) ~[httpcore-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177) ~[httpasyncclient-4.1.2.jar:4.1.2]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81) ~[httpasyncclient-4.1.2.jar:4.1.2]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39) ~[httpasyncclient-4.1.2.jar:4.1.2]
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588) ~[httpcore-nio-4.4.5.jar:4.4.5]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_60]
18:48:21,268 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase] Starting integration tests against an external cluster running elasticsearch [5.1.1] with security
18:48:21,297 INFO  [f.p.e.c.f.t.i.FsCrawlerImplAllParametersIT]  -> Removing existing index [fscrawler_test_add_new_file*]
18:48:21,342 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   --> Launching test [test_add_new_file]
18:48:21,343 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   --> Copying test resources from [/Users/dpilato/Documents/Elasticsearch/dev/fscrawler/target/test-classes/samples/common]
18:48:21,389 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   --> Test resources ready in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]
18:48:21,390 INFO  [f.p.e.c.f.t.i.FsCrawlerImplAllParametersIT]   --> creating crawler for dir [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]
18:48:21,390 INFO  [f.p.e.c.f.t.i.FsCrawlerImplAllParametersIT]   --> starting crawler [fscrawler_test_add_new_file]
18:48:21,393 INFO  [f.p.e.c.f.FsCrawlerImpl] Starting FS crawler
18:48:21,393 INFO  [f.p.e.c.f.FsCrawlerImpl] FS crawler started in watch mode. It will run unless you stop it with CTRL+C.
18:48:21,568 DEBUG [f.p.e.c.f.FsCrawlerImpl] creating fs crawler thread [fscrawler_test_add_new_file] for [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file] every [5s]
11:48:21,569 INFO  [f.p.e.c.f.FsCrawlerImpl] FS crawler started for [fscrawler_test_add_new_file] for [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file] every [5s]
11:48:21,570 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler thread [fscrawler_test_add_new_file] is now running. Run #1...
11:48:21,632 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, folder, ab613db27937c334e755ed5e582e4076
11:48:21,673 DEBUG [f.p.e.c.f.FsCrawlerImpl] indexing [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file] content
11:48:21,674 DEBUG [f.p.e.c.f.f.FileAbstractor] Listing local files from /var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file
11:48:21,678 DEBUG [f.p.e.c.f.f.FileAbstractor] 1 local files found
11:48:21,678 DEBUG [f.p.e.c.f.FsCrawlerImpl] [roottxtfile.txt] can be indexed: [true]
11:48:21,678 DEBUG [f.p.e.c.f.FsCrawlerImpl]   - file: roottxtfile.txt
11:48:21,678 DEBUG [f.p.e.c.f.FsCrawlerImpl] fetching content from [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file],[roottxtfile.txt]
11:48:22,208 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, doc, 8db7c0cd3249e05d70f601782558cab
11:48:22,236 DEBUG [f.p.e.c.f.FsCrawlerImpl] Looking for removed files in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]...
11:48:22,248 DEBUG [f.p.e.c.f.FsCrawlerImpl] Looking for removed directories in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]...
11:48:22,256 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler is going to sleep for 5s
18:48:22,625 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   ---> Waiting up to 20s seconds for some documents in index fscrawler_test_add_new_file
18:48:22,649 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   ---> Waiting up to 20s seconds for 1 documents in index fscrawler_test_add_new_file
18:48:22,653 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expected [1] and got [1] documents in [fscrawler_test_add_new_file]
18:48:22,653 INFO  [f.p.e.c.f.t.i.FsCrawlerImplAllParametersIT]  ---> Adding a copy of roottxtfile.txt
18:48:22,653 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   ---> Waiting up to 20s seconds for 2 documents in index fscrawler_test_add_new_file
18:48:22,656 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,657 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,657 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,657 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,662 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,662 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,662 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,662 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,668 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,668 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,668 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,668 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,677 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,678 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,678 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,678 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,693 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,693 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,693 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,693 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,717 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,717 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,717 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,717 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,760 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,760 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,761 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,761 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,835 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,835 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,835 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,836 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:22,980 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:22,980 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:22,980 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:22,980 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:23,244 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:23,245 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:23,245 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:23,245 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:23,766 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:23,767 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:23,767 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:23,767 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:24,779 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:24,779 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:24,779 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:24,779 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:25,787 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:25,787 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:25,788 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:25,789 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
18:48:26,797 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expecting [2] but got [1] documents in [fscrawler_test_add_new_file]
18:48:26,797 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> content of [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]:
18:48:26,797 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - new_roottxtfile.txt 2017-01-07T17:48:22Z
18:48:26,797 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]          - roottxtfile.txt 2017-01-07T17:48:21Z
11:48:27,256 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler is now waking up again...
11:48:27,256 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler thread [fscrawler_test_add_new_file] is now running. Run #2...
11:48:27,257 DEBUG [f.p.e.c.f.FsCrawlerImpl] indexing [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file] content
11:48:27,258 DEBUG [f.p.e.c.f.f.FileAbstractor] Listing local files from /var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file
11:48:27,258 DEBUG [f.p.e.c.f.f.FileAbstractor] 2 local files found
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl] [new_roottxtfile.txt] can be indexed: [true]
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl]   - file: new_roottxtfile.txt
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl] fetching content from [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file],[new_roottxtfile.txt]
11:48:27,269 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, doc, ea2ac5ad08de0465ff5cae1079bd46
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl] [roottxtfile.txt] can be indexed: [true]
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl]   - file: roottxtfile.txt
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl] fetching content from [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file],[roottxtfile.txt]
11:48:27,290 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, doc, 8db7c0cd3249e05d70f601782558cab
11:48:27,304 DEBUG [f.p.e.c.f.FsCrawlerImpl] Looking for removed files in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]...
11:48:27,316 DEBUG [f.p.e.c.f.FsCrawlerImpl] Looking for removed directories in [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file]...
11:48:27,329 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler is going to sleep for 5s
18:48:27,810 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase]      ---> expected [2] and got [2] documents in [fscrawler_test_add_new_file]
18:48:27,810 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase]   --> Stopping crawler
18:48:27,810 DEBUG [f.p.e.c.f.FsCrawlerImpl] Closing FS crawler [fscrawler_test_add_new_file]
18:48:27,811 DEBUG [f.p.e.c.f.FsCrawlerImpl] FS crawler thread is still running
11:48:27,811 DEBUG [f.p.e.c.f.FsCrawlerImpl] Fs crawler is now waking up again...
11:48:27,811 DEBUG [f.p.e.c.f.FsCrawlerImpl] FS crawler thread [fscrawler_test_add_new_file] is now marked as closed...
18:48:28,311 DEBUG [f.p.e.c.f.FsCrawlerImpl] FS crawler thread is now stopped
18:48:28,312 DEBUG [f.p.e.c.f.FsCrawlerImpl] FS crawler Rest service stopped
18:48:28,313 DEBUG [f.p.e.c.f.FsCrawlerImpl] ES Client Manager stopped
18:48:28,314 INFO  [f.p.e.c.f.FsCrawlerImpl] FS crawler [fscrawler_test_add_new_file] stopped
18:48:28,318 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase] Stopping integration tests against an external cluster
18:48:28,319 INFO  [f.p.e.c.f.t.AbstractFSCrawlerTestCase] Elasticsearch client stopped
18:48:28,319 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] ls -l /var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/.fscrawler
18:48:28,323 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] /var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/.fscrawler/_default
18:48:28,324 DEBUG [f.p.e.c.f.t.AbstractFSCrawlerTestCase] /var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/.fscrawler/fscrawler_test_add_new_file
```

Here is the significant part:

```
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl] [new_roottxtfile.txt] can be indexed: [true]
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl]   - file: new_roottxtfile.txt
11:48:27,259 DEBUG [f.p.e.c.f.FsCrawlerImpl] fetching content from [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file],[new_roottxtfile.txt]
11:48:27,269 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, doc, ea2ac5ad08de0465ff5cae1079bd46
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl] [roottxtfile.txt] can be indexed: [true]
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl]   - file: roottxtfile.txt
11:48:27,278 DEBUG [f.p.e.c.f.FsCrawlerImpl] fetching content from [/var/folders/r_/r14sy86n2zb91jyz1ptb5b4w0000gn/T/junit7823904466616904811/resources/test_add_new_file],[roottxtfile.txt]
11:48:27,290 DEBUG [f.p.e.c.f.FsCrawlerImpl] Indexing in ES fscrawler_test_add_new_file, doc, 8db7c0cd3249e05d70f601782558cab
```

**Comment by babadofar** [comment]
I'm on windows and I get creationTime set to the same as last accessed time , which results in files being mistaken for new.  Perhaps it's more reliable to just use last modified date?

**Comment by dadoonet** [comment]
Probably. I just created a test which proves that for now it has the wrong behavior.
Will try to fix it and then may be I'll ask you if you can run the branch on Windows?

**Comment by dadoonet** [comment]
I see what is happening. Actually it's caused by https://github.com/dadoonet/fscrawler/commit/5a2e856b1b6547c58581e340618a1d6658fc909f#diff-b7b2d003d4e7119b715355589e949d29R490 where I on purpose moved the actual indexing date by 2 seconds.

```java
scanDate = scanDate.minus(2, ChronoUnit.SECONDS);
```

In the context of the test I ran, I'm writing a file on disk at time `T`, launch the test at time `T+1ms` for example.
Then I write the last scan date as `T+1ms-2s` which basically means that the file I wrote at `T` appears newer than the last time we scan the hard disk.

So it's not really a bug. Just that if you start FSCrawler and have a file which has been modified in the last 2 seconds, this file will be indexed twice.
